source/http-boot/pmem_node.md

```diff
@@ -0,0 +1,131 @@
+making the contents available to userspace.
+
+The kernel must include drivers required for PMEM devices support.
+
+## Firmware responsibilities
+
+The firmware is responsible for inserting a PMEM Device Tree node that describes
+the persistent memory region containing the downloaded image, allowing the
+operating system to discover and access it during boot.
+
+### PMEM Representation in Device Tree
+
+According to the PMEM Device Tree bindings, the memory region should:
+
+- be described as volatile memory
+- include the `"pmem-region"` compatible string
+- use start and end addresses that satisfy the alignment requirements imposed
+  by the kernel configuration (page size or 2 MiB)
+
+The following example illustrates a PMEM region inserted into Device Tree by firmware.
+```dts
+/* Example PMEM Device Tree node */
+pmem@40200000 {
+        reg = <0x00 0x40200000 0x00 0x24600000>;
+        volatile;
+        compatible = "pmem-region";
+};
+```
+
+### PMEM Memory Region and EFI memory map
+
+It is recommended that firmware exclude the PMEM memory region from the EFI memory
+map handed over to the OS. If the region is instead included in the EFI memory map
+and only marked as reserved in Device Tree, then depending on the OS kernel version
+and configuration the PMEM driver may fail to reserve the region for a namespace
+and instantiate a `/dev/pmemX` device.
+
+#### Details for Linux 5.x and 6.10 kernel versions
+
+If the `CONFIG_ZONE_DEVICE` and `CONFIG_SPARSEMEM` kernel options are enabled, the PMEM driver
+invokes `devm_memremap_pages()` instead of `devm_memremap()`. While
+`devm_memremap()` works correctly whether the memory region is omitted from the
+EFI memory map or marked as reserved, `devm_memremap_pages()` only succeeds when
+the region is omitted.
+
+The following kernel messages illustrate the failure observed when the PMEM
+driver is unable to reserve the memory region:
+```
+[  111.743083] [  T517] nd_pmem namespace0.0: could not reserve region [mem 0x48000000-0x67ffffff]
+[  111.743211] [  T517] nd_pmem namespace0.0: probe with driver nd_pmem failed with error -16
+```
+
+A kernel fix addressing this issue has been proposed and is under review:
+[Don't call __add_pages() for reserved memory](https://gitlab.com/ilias.apalodimas/net-next/-/commit/0e1b2d01511e59008ef21c3c84cd347780cd0a51)
+
+## OS responsibilities
+
+With a correct PMEM Device Tree node definition and all required drivers included
+in the Linux kernel, the OS instantiates a `/dev/pmemX` device. As needed,
+partitions contained within the PMEM region can then be mounted by the installer
+or live environment. Support depends on kernel configuration and the contents of
+the installer or live image.
+
+### Linux Kernel Support Requirements
+
+The following kernel configuration options are required to support PMEM regions
+described via Device Tree:
+```
+CONFIG_ARM64_PMEM
+CONFIG_LIBNVDIMM
+CONFIG_BLK_DEV_PMEM
+CONFIG_OF_PMEM
```

Comment:
DEV_DAX_PMEM

I dont think  this is *needed* but it's good to have in case raw device access is needed